consensus/src/quorum_store/types.rs

```diff
@@ -7,9 +7,55 @@ use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
 use aptos_types::{transaction::SignedTransaction, PeerId};
 use bcs::to_bytes;
 use serde::{Deserialize, Serialize};
-use std::mem;
+use std::{
+    cmp::Ordering,
+    fmt::{Display, Formatter},
+    mem,
+};
+
+#[derive(
+    Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash, CryptoHasher, BCSCryptoHash,
+)]
+pub struct BatchId {
+    pub id: u64,
+    pub nonce: u64,
```

Comment:
add a comment to explain nonce is some random number generated every time after recovery